test/k8s_tests/domain_mapping_e2e_test.go

```diff
@@ -0,0 +1,126 @@
+	utilst "github.com/dana-team/container-app-operator/test/k8s_tests/utils"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	knativev1beta1 "knative.dev/serving/pkg/apis/serving/v1beta1"
+)
+
+var _ = Describe("Validate DomainMapping functionality", func() {
+
+	It("Should create DomainMapping when Creating a Capp instance", func() {
+		By("Creating a capp with a route")
+		routeCapp := mock.CreateBaseCapp()
+		routeHostname := utilst.GenerateRouteHostname()
+		routeCapp.Spec.RouteSpec.Hostname = routeHostname
+		utilst.CreateCapp(k8sClient, routeCapp)
+		By("Checking if the domainMapping was created successfully")
+		domainMapping := &knativev1beta1.DomainMapping{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      routeHostname,
+				Namespace: routeCapp.Namespace,
+			},
+		}
+		Eventually(func() bool {
+			return utilst.DoesResourceExist(k8sClient, domainMapping)
+		}, TimeoutCapp, CappCreationInterval).Should(BeTrue(), "Should find a resource.")
+	})
+
+	It("Should create DomainMapping with secret when Creating an https Capp instance", func() {
+		By("Creating a secret")
+		secretName := utilst.GenerateSecretName()
+		secret := v1.Secret{
+			TypeMeta: metav1.TypeMeta{},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      secretName,
+				Namespace: mock.NsName,
+			},
+			Type: "Opaque",
+			Data: map[string][]byte{"extra": []byte("YmFyCg==")},
+		}
+		utilst.CreateSecret(k8sClient, secret)
+		By("Creating an https capp")
+		httpsCapp := mock.CreateBaseCapp()
+		routeHostname := utilst.GenerateRouteHostname()
+		httpsCapp.Spec.RouteSpec.Hostname = routeHostname
+		httpsCapp.Spec.RouteSpec.TlsEnabled = true
+		httpsCapp.Spec.RouteSpec.TlsSecret = secretName
+		utilst.CreateCapp(k8sClient, httpsCapp)
+		By("Checking if the secret reference exists at the domainMapping")
+		Eventually(func() string {
+			domainMapping := utilst.GetDomainMapping(k8sClient, routeHostname, httpsCapp.Namespace)
+			return domainMapping.Spec.TLS.SecretName
+		}, TimeoutCapp, CappCreationInterval).Should(Equal(secretName))
+	})
+
+	It("Should delete DomainMapping when deleting a Capp instance", func() {
+		By("Creating a capp with a route")
+		routeCapp := mock.CreateBaseCapp()
+		routeHostname := utilst.GenerateRouteHostname()
+		routeCapp.Spec.RouteSpec.Hostname = routeHostname
+		createdCapp := utilst.CreateCapp(k8sClient, routeCapp)
+		By("Checking if the domainMapping was created successfully")
+		domainMapping := &knativev1beta1.DomainMapping{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      routeHostname,
+				Namespace: routeCapp.Namespace,
+			},
+		}
+		Eventually(func() bool {
+			return utilst.DoesResourceExist(k8sClient, domainMapping)
+		}, TimeoutCapp, CappCreationInterval).Should(BeTrue(), "Should find a resource.")
+		CreatedDomainMapping := utilst.GetDomainMapping(k8sClient, routeHostname, routeCapp.Namespace)
+		By("Deleting the capp instance")
```

Comment:
I don't think  you should separate the base `CRUD` logic to different `Its` because you repeating the creation of Capp and the profit of concurrency is minimal. The edge cases should be in different  `Its`.